pkg/api/types.go

```diff
@@ -1205,6 +1221,13 @@ type KeyToPath struct {
 	Mode *int32
 }
 
+// Local represents a directory with specific topology contraints
+type LocalVolumeSource struct {
+	// The full path to the volume on the node
+	// For alpha, this path must be a file-based volume
+	Path string
```

Comment:
I totally agree that `pv.spec.volumeType` and `pvc.spec.volumeType: file or block` makes sense and is the right way to go (in a separate PR/proposal), but I am commenting something else.

Right now, `LocalVolumeSource` covers only filesystems mounted outside of Kubernetes. To some extent it's very similar to HostPath. `LocalVolumeSource.PathToMountedDirectory` is the only thing we need here.

In future, we want `LocalVolumeSource` to cover also block devices, mounted by Kubernetes. This is similar to GCE PD or AWS EBS, just without attach/detach, there already is a device in /dev. We will need `LocalVolumeSource.PathToDevice` (which is IMO something different than `PathToMountedDirectory`) and `FSType`. Again, overloading `LocalVolumeSource.Path` for both paths is bad, its semantics changes based on what kind of local volume it is. That's why I think two structures (directory / block) are better than one. It's clear what `Path` means and it's clear that fsType applies only to local block volumes.